mlir/lib/Dialect/Affine/IR/AffineOps.cpp

```diff
@@ -443,9 +444,15 @@ bool mlir::affine::isValidSymbol(Value value, Region *region) {
   if (matchPattern(defOp, m_Constant(&operandCst)))
     return true;
 
-  // Affine apply operation is ok if all of its operands are ok.
-  if (auto applyOp = dyn_cast<AffineApplyOp>(defOp))
-    return applyOp.isValidSymbol(region);
+  // `Pure` operation that its operands are the valid symbolic identifiers.
+  if (isPure(defOp)) {
+    bool operandVaildSymbol =
+        llvm::all_of(defOp->getOperands(), [&](Value operand) {
+          return affine::isValidSymbol(operand, region);
+        });
+    if (operandVaildSymbol)
```

Comment:
Nit: 
```
if (llvm::all_of(...)
 ...
```
Your comment above already makes it clear - no need of the separate variable.